keyvi/tests/cpp/stringdistance/distance_matrix_test.cpp

```diff
@@ -0,0 +1,48 @@
+//
+// keyvi - A key value store.
+//
+// Copyright 2015 Hendrik Muhs<hendrik.muhs@gmail.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/*
+ * distance_matrix_test.cpp
+ *
+ *  Created on: Jan 17, 2017
+ *      Author: konstantin.s
+ */
+
+#include <boost/test/unit_test.hpp>
+#include "stringdistance/distance_matrix.h"
+
+namespace keyvi {
+namespace stringdistance {
+
+BOOST_AUTO_TEST_SUITE( DistanceMatrixTests )
+
+void TestResizeOverflow(DistanceMatrix& matrix) {
+  const size_t new_rows_number = 1LL + (1LL << 32LL);
+  matrix.EnsureRowCapacity(new_rows_number);
```

Comment:
ok, now I am getting what you want you mean with overflow.

This is rather esoteric. The distance matrix is used by the fuzzy matching code, such a large matrix is unrealistic. I wonder how you executed it? They payload alone is 16 (uint32) or even 32 (uint64) GB in ram. 

Sorry, we can not have such a unit test, this never executes in travis. It's also not very useful. 

It would also mean that a single key has the length of 4GB. I think that's not realistic as well. Just as a side node: I looked up redis: it has a max key length of 512MB.